metis-enrichment/metis-enrichment-common/src/main/java/eu/europeana/enrichment/api/internal/AbstractReferenceTerm.java

```diff
@@ -1,40 +1,40 @@
 package eu.europeana.enrichment.api.internal;
 
-
 import java.net.URL;
 import java.util.Objects;
 
-public abstract class AbstractReferenceTerm implements ReferenceTerm{
+/**
+ * This class is a basic implementation of {@link ReferenceTerm} that leaves the details of the
+ * candidate types unimplemented.
+ */
+public abstract class AbstractReferenceTerm implements ReferenceTerm {
 
   private final URL reference;
 
-  public AbstractReferenceTerm(URL reference){
+  public AbstractReferenceTerm(URL reference) {
     this.reference = reference;
   }
 
   @Override
-  public URL getReference(){
+  public URL getReference() {
     return reference;
   }
 
   @Override
-  public boolean equals(Object other){
-    if(other == this){
+  public final boolean equals(Object otherObject) {
```

Comment:
Why final here, were are enforcing equals/hashcode in the hierarchy which I believe we should not